utils/vlog/tests/dsp_in_registered/golden.model.xml

```diff
@@ -1,12 +1,12 @@
 <models xmlns:xi="http://www.w3.org/2001/XInclude">
   <xi:include href="../dsp_combinational/dsp_combinational.model.xml" xpointer="xpointer(models/child::node())"/>
-  <!-- https://docs.verilogtorouting.org/en/latest/tutorials/arch/timing_modeling/#mixed-sequential-combinational-block -->
+  <xi:include href="../fig42-dff/dff.model.xml" xpointer="xpointer(models/child::node())"/>
   <model name="dsp_in_registered">
     <input_ports>
-      <port name="a" combinational_sink_ports="out" clock="clk"/>
-      <port name="b" combinational_sink_ports="out" clock="clk"/>
-      <port name="clk" is_clock="1"/>
-      <port name="m" combinational_sink_ports="out" clock="clk"/>
+      <port combinational_sink_ports="out" name="a"/>
```

Comment:
Are the clock attributes no longer needed because the dff model captures that dependency now?

Reply:
yes, DFF defines this https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/utils/vlog/tests/fig42-dff/golden.model.xml

Reply:
FYI - A model never includes another model. Models are low level primitives.

**However**, we want to support instantiating a whole module or to instantiate just the parts. An example would be something like;
```verilog
module ABC(...);
    PART_A a(...);
    PART_B b(...);
endmodule;
```
Then people can either go;
```verilog
// Instantiate whole module in one go...
ABC abc(...);
/// ---- or ----
// Instantiate parts of module in a compatible configuration to be placed into the ABC module
PART_A a(...);
PART_B b(...);
```

Then the `pb_type.xml` ends up looking like this;
```xml
<pb_type name="module">
  <mode name="ALL_AT_ONCE">
    <pb_type name="whole_module" blif_model=".subckt module">
    </pb_type>
  </mode>
  <mode name="INDIVIDUAL_PARTS">
    <xi:include href="part_a.pb_type.xml" />
    <xi:include href="part_b.pb_type.xml" />
    <interconnect>
       ....
    </interconnect>
  </mode>
</pb_type>
```

And as the "whole" mode is a subckt, it needs a `<model>`.

Am I making any sense?

Reply:
Poke.

Reply:
I see your point here, but isn't it over complicating the whole flow?

I think we can simply disable emiting include directives in model generator and leave it with with only top level (primitive) definition. All the parts that are included, were actually generated as dependencies for the currently processing one, so we already have their models definition. So a user can either instantiate them, or the "whole" one without "includes".

Does that make sense?

Reply:
@kgugala Ultimately the `model.xml` should be **only** includes of other models and top level models.

Reply:
Just confirming,

### Valid

```xml
<models>
  <xi:include href="../common_slice/common_slice.model.xml" xpointer="xpointer(models/child::node())" />
  <model name="blah">
  </model>
</models>
```

### Never valid

```xml
<models>
  <model name="a">
    <xi:include href="../common_slice/common_slice.model.xml" xpointer="xpointer(models/child::node())" />
  </model>
  <model name="b">
  </model>
</models>
```

Reply:
so the model here has correct form:
```xml
<models xmlns:xi="http://www.w3.org/2001/XInclude">
  <xi:include href="../dsp_combinational/dsp_combinational.model.xml" xpointer="xpointer(models/child::node())"/>
  <xi:include href="../fig42-dff/dff.model.xml" xpointer="xpointer(models/child::node())"/>
  <model name="dsp_in_registered">
    <input_ports>
      <port combinational_sink_ports="out" name="a"/>
      <port combinational_sink_ports="out" name="b"/>
      <port is_clock="1" name="clk"/>
      <port clock="clk" combinational_sink_ports="out" name="m"/>
    </input_ports>
    <output_ports>
      <port name="out"/>
    </output_ports>
  </model>
</models>
```
It only misses the info about clock attributes

Reply:
I added ASSOC_CLOCK attribute and fixed vlog_to_model script so the attribute is actually used. Now the model is emitted with clock attributes